packages/discovery-react-components/src/components/DiscoverySearch/types.ts

```diff
@@ -17,6 +18,14 @@ export interface DocumentProviderProps
   extracted_metadata?: JsonObject;
 }
 
+/**
+ * Document data for PDFSource
+ */
+export type PDFSourceDocument = {
+  __type: 'pdf';
```

Comment:
Is there a reason to use `__`? I prefer simply:

```suggestion
   type: 'pdf';
```